src/acl/Asn.cc

```diff
@@ -96,9 +81,14 @@ class ASState
 CBDATA_CLASS_INIT(ASState);
 
 /** entry into the radix tree */
-struct rtentry_t {
+class rtentry_t
+{
+    MEMPROXY_CLASS(rtentry_t);
+public:
+    rtentry_t() { memset(&e_nodes, 0, sizeof(e_nodes)); }
+
     struct squid_radix_node e_nodes[2];
-    as_info *e_info;
+    std::list<int> e_info;
```

Comment:
If we want to use MEMPROXY_CLASS() for this class, then we should be pooling these `int` allocations as well, right?

Reply:
We could, but the two do not need to be identical.

Reply:
I do not know what you mean by "identical" in this context, but adding pooling for rtentry_t is inconsistent with changing a primary rtentry_t component to always do unpooled memory allocations.

This is not my _primary_ concern, but if this PR survives, it should avoid adding these unpooled memory allocations, especially since the same PR tries to add pooling in other areas. I see no justification for this waste and inconsistency.

Rule of thumb: If you add an std::list, you probably need to add pooling for its member wrappers.

Reply:
Done in 74bad9504

Reply:
AFAICT, 74bad95 misses the case this change request was attached to.